ui/src/app/components/object-explorer/object-explorer.component.ts

```diff
@@ -182,4 +199,141 @@ export class ObjectExplorerComponent implements OnInit {
   setSpannerTab() {
     this.selectedIndex = 1
   }
+
+  checkDropSelection(): boolean {
+    let countByCategory = this.countSelectionByCategory()
+    if (countByCategory.eligibleForDrop != 0) {
+      return true
+    }
+    return false
+  }
+
+  checkRestoreSelection(): boolean {
+    let countByCategory = this.countSelectionByCategory()
+    if (countByCategory.eligibleForRestore != 0) {
+      return true
+    }
+    return false
+  }
+
+  countSelectionByCategory(): {eligibleForDrop: number, eligibleForRestore: number} {
+    const values = this.checklistSelection.selected
+    let eligibleForDrop = 0
+    let eligibleForRestore = 0
+    values.forEach((flatNode) => {
+      if (!flatNode.isDeleted) {
+        eligibleForDrop += 1
+      } else {
+        eligibleForRestore += 1
+      }
+    })
+    return {eligibleForDrop: eligibleForDrop, eligibleForRestore: eligibleForRestore}
+  }
+
+  dropSelected() {
+    //selected values to be dropped
+    const values = this.checklistSelection.selected
+    var tablesWithState: ITableStateList = {
+      TableList: []
+    };
+    
+    values.forEach((flatNode) => {
+      if (flatNode.id != "" && flatNode.id.startsWith("t")) {
```

Comment:
Rather than checking if the id starts with "t" it would be better to compare the flatNode.Type with ObjectExplorerNodeType.table IMO

Reply:
Done